runbot_gitlab_minimal/models/runbot_repo.py

```diff
@@ -0,0 +1,91 @@
+        if os.path.isdir(os.path.join(repo.path, 'refs')) and repo.uses_gitlab:
+            repo.git(['fetch', '-p', 'origin',
+                      '+refs/merge-requests/*/head:refs/pull/*'])
+        return super(RunbotRepo, self).update_git(cr, uid, repo,
+                                                  context=context)
+
+    def github(self, cr, uid, ids, url, payload=None, ignore_errors=False,
+               context=None):
+        """Translate each request from github format to gitlab format"""
+        for repo in self.browse(cr, uid, ids, context=context):
+            is_url_merge = False
+            if not repo.token:
+                continue
+            if not repo.uses_gitlab:
+                return (super(RunbotRepo, self)
+                        .github(cr, uid, [repo.id], url, payload=payload,
+                                ignore_errors=ignore_errors, context=context))
+            else:
+                try:
+                    match_object = re.search('([^/]+)/([^/]+)/([^/.]+(.git)?)',
+                                             repo.base)
+                    if match_object:
+                        project_name = (match_object.group(2) + '/' +
+                                        match_object.group(3))
+                        url = url.replace(':owner', match_object.group(2))
+                        url = url.replace(':repo', match_object.group(3))
+                        url = 'https://%s/api/v3%s' % (match_object.group(1),
+                                                       url)
+                        url = url.replace('/repos/', '/projects/')
+                        url = url.replace(project_name,
+                                          urllib.quote(project_name, safe=''))
+                        if '/pulls/' in url:
+                            urls = url.split('/pulls/')
+                            url = urls[0] + '/merge_requests?iid=' + urls[1]
+                            is_url_merge = True
+                        session = requests.Session()
+                        session.auth = (repo.token, 'x-oauth-basic')
+                        session.headers.update({'PRIVATE-TOKEN': repo.token})
+                        if payload:
+                            if payload['state'] == 'pending':
+                                payload['state'] = 'running'
+                            if payload['state'] in ('error', 'failure'):
+                                payload['state'] = 'failed'
+                            response = session.post(url, data=payload)
+                        else:
+                            response = session.get(url)
+                        response.raise_for_status()
+                        json = response.json()
+                        if is_url_merge:
+                            json = json[0]
+                            json['head'] = {'ref': json['target_branch']}
+                            json['base'] = {'ref': json['source_branch']}
```

Comment:
Please, review that runbot is not computing the branch base using git directly instead of use this value.